### DISCUSSION TITLE
PostgreSQL incorrect statement separation

### DISCUSSION THREAD

**Discussion by vpetryaev** [dbeaver/dbeaver#35106] [Q&A] [answered]
### Description

DBEaver doesn't separate statements correctly after CREATE PROCEDURE ... LANGUAGE SQL BEGIN ATOMIC ...

### DBeaver Version

Community edition Version 24.1.1.202406231636

### Operating System

Ubuntu 20.04

### Database and driver

Postgresql 15
Driver 42.7.2

### Steps to reproduce

Paste this SQL code, and select all for execute at once:
```
CREATE TABLE test_table(id int, string varchar);
CREATE PROCEDURE test_proc(testdata test_table[])
LANGUAGE SQL
BEGIN ATOMIC
    INSERT INTO test_table (id, string)
    SELECT id, string from unnest(testdata);
END;
CALL test_proc(testdata => ARRAY[(1, 'one'), (2, 'two')]::test_table[]);

```

Result is SQL Error [42601]: ERROR: cannot insert multiple commands into a prepared statement

### Additional context

Using psql, no issues:
```
test=# CREATE TABLE test_table(id int, string varchar);
CREATE TABLE
test=# CREATE PROCEDURE test_proc(testdata test_table[])
test-# LANGUAGE SQL
test-# BEGIN ATOMIC
test-#     INSERT INTO test_table (id, string)
test-#     SELECT id, string from unnest(testdata);
test-# END;
CREATE PROCEDURE
test=# CALL test_proc(testdata => ARRAY[(1, 'one'), (2, 'two')]::test_table[]);
CALL

```
In case of sequential execution of CREATE TABLE and CALL in DBEaver - also no issues.

**Comment by E1izabeth** [comment]
Thank you for bug report.

**Comment by E1izabeth** [comment, marked as answer]
Please try executing it as script.
When you select something in SQL Editor and press Execute (Ctrl+Enter), the selected part of text will be send as a single statement to the database for execution. If you want to execute several statements, this way of execution doesn't fit.
